spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/amqp/RabbitMetrics.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.metrics.amqp;
+
+import java.util.Collections;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.ConnectionFactory;
+import com.rabbitmq.client.impl.AbstractMetricsCollector;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.binder.MeterBinder;
+
+/**
+ * A {@link MeterBinder} for RabbitMQ Java Client metrics.
+ *
+ * @author Arnaud Cogoluègnes
+ * @since 2.0.0
+ */
+public class RabbitMetrics implements MeterBinder {
+
+	private final Iterable<Tag> tags;
+
+	private final ConnectionFactory connectionFactory;
+
+	public RabbitMetrics(ConnectionFactory connectionFactory) {
+		this(connectionFactory, Collections.emptyList());
+	}
+
+	public RabbitMetrics(ConnectionFactory connectionFactory, Iterable<Tag> tags) {
+		this.connectionFactory = connectionFactory;
+		this.tags = tags;
+	}
+
+	@Override
+	public void bindTo(MeterRegistry registry) {
+		this.connectionFactory.setMetricsCollector(new MicrometerMetricsCollector(registry, "rabbitmq", this.tags));
+	}
+
+	/**
+	 * Micrometer implementation of {@link com.rabbitmq.client.MetricsCollector}.
+	 */
+	public static class MicrometerMetricsCollector extends AbstractMetricsCollector {
```

Comment:
Micrometer support isn't available in RabbitMQ Java Client 5.0.x (which is the version Spring Boot 2.0 depends on AFAIR). It was introduced in 4.3.0 and pushed upstream in 5.1.0. Spring Boot can choose to depend on 5.1.1 (latest) to re-use the existing `MicrometerMetricsCollector`, even though Spring AMQP depends on 5.0.0.
`MicrometerMetricsCollector` code is straightforward though, the parent class handle the heavy-lifting.

Reply:
@acogoluegnes Using 5.1.x seems like the best option purely from Boot's perspective. Is 5.1 backwards compatible with 5.0 or is it likely to cause problems for Spring AMQP do you know?

/cc @garyrussell @artembilan

Reply:
I just tested master against 5.1.1 with no problems.

Reply:
I think we can't upgrade Spring AMQP 2.0.x to the AMQP Client 5.1.x, but Spring Boot 2.0 still can do that and therefore replace Spring AMQP's one.

Reply:
@wilkinsona 5.1 is backwards compatible with 5.0, RabbitMQ Java Client uses semantic versioning as of 4.0.0. Thanks for testing out @garyrussell.

@wilkinsona As @artembilan and you suggested, you can use 5.1.1 in Spring Boot, it should cause any trouble to Spring AMQP.

Reply:
@artembilan Could we push towards a Spring AMQP 2.1.x? Or is this too much of a support burden?

Reply:
There is nothing yet for Spring AMQP 2.1. 
More over we only talk about a compatible dependency. So, if you don't use Spring Boot but only Spring AMQP, you should override dependency manually. That's all.
Otherwise with the Spring Boot everything is transparent for you.
And yes: that is really too much to support...

Reply:
If the version police will allow it, I wouldn't mind changing to 5.1.x in 2.0.

Reply:
I guess I'm (one of) the version police :). I have no objections to doing that.